### ISSUE TITLE
serve property in webpack.config.js is treated as invalid

### ISSUE THREAD

**Issue by kleinph** [webpack/webpack#7306]
# Bug report

**What is the current behavior?**
A _webpack.config.js_ containing a `serve` property (to configure https://github.com/webpack-contrib/webpack-serve) is treated as invalid.

**If the current behavior is a bug, please provide the steps to reproduce.**
Use a _webpack.config.js_ which contains the `serve` property and start webpack.

**What is the expected behavior?**
A _webpack.config.js_ containing a `serve` property should not be treated as invalid. The `devServer` property is also not treated as invalid.

**Other relevant information:**
webpack version:  4.8.3
Node.js version: v10.1.0
Operating System: Linux 4.16.8
Additional tools: webpack-serve 1.0.2

**Comment by montogeek** [comment]
cc @shellscape

**Comment by evilebottnawi** [comment]
Strange for me all works fine
@kleinph can you provide your configuration

**Comment by kleinph** [comment]
```
npm run build

> danube@ build /home/user1/webapp
> webpack --mode production

Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration has an unknown property 'serve'. These properties are valid:
   object { mode?, amd?, bail?, cache?, context?, dependencies?, devServer?, devtool?, entry?, externals?, loader?, module?, name?, node?, output?, optimization?, parallelism?, performance?, plugins?, profile?, recordsInputPath?, recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, stats?, target?, watch?, watchOptions? }
   For typos: please correct them.
   For loader options: webpack >= v2.0.0 no longer allows custom properties in configuration.
     Loaders should be updated to allow passing options via loader options in module.rules.
     Until loaders are updated one can use the LoaderOptionsPlugin to pass these options to the loader:
     plugins: [
       new webpack.LoaderOptionsPlugin({
         // test: /\.xxx$/, // may apply this only for some modules
         options: {
           serve: …
         }
       })
     ]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! danube@ build: `webpack --mode production`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the danube@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user1/.npm/_logs/2018-05-15T10_14_28_664Z-debug.log
```

**Comment by evilebottnawi** [comment]
Yep, it is bug, feel free to Send PR :+1:

**Comment by evilebottnawi** [comment]
It is very easy :+1:

**Comment by JRJurman** [comment]
is there a release with this fix?

**Comment by iwazaru** [comment]
@JRJurman Fixed in [4.9.0](https://github.com/webpack/webpack/releases/tag/v4.9.0)
